### PR TITLE
Convert to GDA docstring to doctest

### PR DESCRIPTION
**Pull request by yashk2810**

### CODE REVIEW

**Comment by yashk2810** [comment]
I fixed the line break issue

image

File: jax/experimental/global_device_array.py [hunk 1]
```diff
@@ -392,13 +407,19 @@ def from_callback(cls, global_shape: Shape, global_mesh: pxla.Mesh,
...
-    Example::
+    Example:
 
-      global_input_shape = (8, 2)
-      global_input_data = np.arange(prod(global_input_shape)).reshape(global_input_shape)
-      def cb(index):
-        return global_input_data[index]
-      gda = GlobalDeviceArray.from_callback(global_input_shape, global_mesh, mesh_axes, cb)
+      >>> from jax.experimental.maps import Mesh
+      >>> import numpy as np
```
Comment (skye) [inline jax/experimental/global_device_array.py:413]: Add a line break after imports (here and below)

File: jax/experimental/global_device_array.py [hunk 2]
```diff
@@ -392,13 +407,19 @@ def from_callback(cls, global_shape: Shape, global_mesh: pxla.Mesh,
...
+      >>> global_input_shape = (8, 8)
+      >>> mesh_axes = ['x', 'y']
+      >>> global_mesh = global_mesh = Mesh(np.array(jax.devices()).reshape(2, 4), ('x', 'y'))
+      >>> global_input_data = np.arange(prod(global_input_shape)).reshape(global_input_shape)
+      >>> def cb(index):
```
Comment (skye) [inline jax/experimental/global_device_array.py:418]: Add line breaks before and after function definitions for readability (here and below)

**Review by skye — APPROVED** [review]
Looks great! Just a few minor comments, otherwise LGTM

File: jax/experimental/global_device_array.py (hunk 2 above)
Comment (yashk2810) [reply jax/experimental/global_device_array.py:420]: DOne

File: jax/experimental/global_device_array.py (hunk 1 above)
Comment (yashk2810) [reply jax/experimental/global_device_array.py:413]: Done

**Pull request merged**